torchvision/datapoints/_dataset_wrapper.py

```diff
@@ -288,30 +290,40 @@ def wrapper(idx, sample):
         if not target:
             return image, dict(image_id=image_id)
 
+        spatial_size = tuple(F.get_spatial_size(image))
         batched_target = list_of_dicts_to_dict_of_lists(target)
+        target = {}
 
-        batched_target["image_id"] = image_id
+        if "image_id" in target_keys:
+            target["image_id"] = image_id
 
-        spatial_size = tuple(F.get_spatial_size(image))
-        batched_target["boxes"] = F.convert_format_bounding_box(
-            datapoints.BoundingBox(
-                batched_target["bbox"],
-                format=datapoints.BoundingBoxFormat.XYWH,
-                spatial_size=spatial_size,
-            ),
-            new_format=datapoints.BoundingBoxFormat.XYXY,
-        )
-        batched_target["masks"] = datapoints.Mask(
-            torch.stack(
-                [
-                    segmentation_to_mask(segmentation, spatial_size=spatial_size)
-                    for segmentation in batched_target["segmentation"]
-                ]
-            ),
-        )
-        batched_target["labels"] = torch.tensor(batched_target["category_id"])
+        if "boxes" in target_keys:
+            target["boxes"] = F.convert_format_bounding_box(
+                datapoints.BoundingBox(
+                    batched_target["bbox"],
+                    format=datapoints.BoundingBoxFormat.XYWH,
+                    spatial_size=spatial_size,
+                ),
+                new_format=datapoints.BoundingBoxFormat.XYXY,
+            )
+
+        if "masks" in target_keys:
+            target["masks"] = datapoints.Mask(
+                torch.stack(
+                    [
+                        segmentation_to_mask(segmentation, spatial_size=spatial_size)
+                        for segmentation in batched_target["segmentation"]
+                    ]
+                ),
+            )
```

Comment:
If we don't do that by default anymore, is this going to bias the benchmarks in favour of v2 (because the v1 path still does it no matter what we pass)?

Reply:
We can setup the benchmarks to always decode the masks and drop them after. With this we have comparable results, while users still get the best performance.